src/coremidi.cs

```diff
@@ -228,5 +253,172 @@ interface MidiCISession
+		IntPtr Constructor (MidiCIDiscoveredNode discoveredNode, Action handler, MidiCISessionDisconnectHandler disconnectHandler);
+
+		[Mac (10, 16), iOS (14, 0)]
+		[Export ("sendProfile:onChannel:profileData:")]
+		bool SendProfile (MidiCIProfile profile, byte channel, NSData profileSpecificData);
+
+		[Mac (10, 16), iOS (14, 0)]
+		[Export ("deviceInfo")]
+		MidiCIDeviceInfo DeviceInfo { get; }
+
+		[Mac (10, 16), iOS (14, 0)]
+		[Export ("maxSysExSize")]
+		NSNumber MaxSysExSize { get; }
+
+		[Mac (10, 16), iOS (14, 0)]
+		[Export ("maxPropertyRequests")]
+		NSNumber MaxPropertyRequests { get; }
+
+		[Internal]
+		[Mac (10, 16), iOS (14, 0)]
+		[Export ("midiDestination")]
+		MidiObjectRef _MidiDestination { get; }
+
+		[Mac (10, 16), iOS (14, 0)]
+		MidiEndpoint MidiDestination {
+			[Wrap ("new MidiEndpoint (_MidiDestination)")]
+			get;
+		}
+
+		[Mac (10, 16), iOS (14, 0)]
+		[NullAllowed, Export ("profileSpecificDataHandler", ArgumentSemantic.Copy)]
+		MidiCIProfileSpecificDataHandler ProfileSpecificDataHandler { get; set; }
 	}
+
+	[Mac (10, 16), iOS (14, 0)]
+	[BaseType (typeof (NSObject), Name="MIDICIDeviceInfo")]
+	[DisableDefaultCtor]
+	interface MidiCIDeviceInfo : NSSecureCoding
+	{
+		[Export ("manufacturerID")]
+		NSData ManufacturerID { get; }
+
+		[Export ("family")]
+		NSData Family { get; }
+
+		[Export ("modelNumber")]
+		NSData ModelNumber { get; }
+
+		[Export ("revisionLevel")]
+		NSData RevisionLevel { get; }
+
+		[Internal]
+		[Export ("midiDestination")]
+		MidiObjectRef _MidiDestination { get; }
+
+		MidiEndpoint MidiDestination {
+			[Wrap ("new MidiEndpoint (_MidiDestination)")]
+			get;
+		}
+
+		[Internal]
+		[Export ("initWithDestination:manufacturer:family:model:revision:")]
+		IntPtr Constructor (MidiObjectRef midiDestination, NSData manufacturer, NSData family, NSData modelNumber, NSData revisionLevel);
+
+		[Wrap ("this (midiDestination.Handle, manufacturer, family, modelNumber, revisionLevel)")]
+		IntPtr Constructor (MidiEndpoint midiDestination, NSData manufacturer, NSData family, NSData modelNumber, NSData revisionLevel);
+	}
+
+	[Mac (10, 16), iOS (14, 0)]
+	[BaseType (typeof (NSObject), Name="MIDICIDiscoveredNode")]
+	[DisableDefaultCtor]
+	interface MidiCIDiscoveredNode
+	{
+		[Internal]
+		[Export ("destination")]
+		MidiObjectRef _Destination { get; }
+
+		MidiEndpoint Destination { 
+			[Wrap ("new MidiEndpoint (_Destination)")]
+			get;
+		}
+
+		[Export ("deviceInfo")]
+		MidiCIDeviceInfo DeviceInfo { get; }
+
+		[Export ("supportsProfiles")]
+		bool SupportsProfiles { get; }
+
+		[Export ("supportsProperties")]
+		bool SupportsProperties { get; }
+
+		[Export ("maximumSysExSize")]
+		NSNumber MaximumSysExSize { get; }
+	}
+
+	delegate void MidiCIDiscoveryResponseDelegate (MidiCIDiscoveredNode[] arg0);
+
+	[Mac (10, 16), iOS (14, 0)]
+	[BaseType (typeof (NSObject), Name="MIDICIDiscoveryManager")]
+	interface MidiCIDiscoveryManager
```

Comment:
`[DisableDefaultCtor]` ?
in most case a `sharedInstance` means it's not meant to be user-created